web/profiles/custom/os2loop/modules/os2loop_upvote/os2loop_upvote.module

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/**
+ * @file
+ * The module file for os2loop_upvote.
+ */
+
+use Drupal\Core\Cache\Cache;
+
+/**
+ * Add number of upvotes to flag.
+ */
+function os2loop_upvote_preprocess_flag(array &$variables) {
+  $comment = $variables['flaggable'];
+  $flag_service = Drupal::service('flag.count');
+  $flag_counts = $flag_service->getEntityFlagCounts($comment);
+  if (isset($flag_counts['os2loop_upvote_upvote_button'])) {
+    $variables['upvotes'] = intval($flag_counts['os2loop_upvote_upvote_button']);
+  }
+  // Clear cache, so the UI reflects changes in top comment.
+  Cache::invalidateTags($comment->getCacheTags());
+}
+
+/**
+ * Find the top comment.
+ *
+ * If there is a comment makred as "correct answer", this is it.
+ * If there is no "correct answer", the top comment will be the most upvoted.
+ */
+function os2loop_upvote_preprocess_field(array &$variables, $hook) {
```

Comment:
Consider creating a service, `FlagHelper`, say, for this and then just delegate from hook implementations to the helper, e.g.

```
function os2loop_upvote_preprocess_flag(array &$variables) {
  Drupal::service('os2loop_upvote.flag_helper')->preprocessFlag(array &$variables);
}
```

Creating a service makes it easier to maintain the hook implementation as we can use OOP and other fancy stuff rather than relying on the global scope. 

See https://github.com/os2loop/os2loop/blob/feature/LOOP-748-documents-and-document-collections-no-menu/web/profiles/custom/os2loop/modules/os2loop_documents/os2loop_documents.module for an example.